core/src/dbs/executor.rs

```diff
@@ -83,7 +87,19 @@ impl<'a> Executor<'a> {
 					let _ = txn.cancel().await;
 				} else {
 					let r = match txn.complete_changes(false).await {
-						Ok(_) => txn.commit().await,
+						Ok(_) => {
+							match txn.commit().await {
+								Ok(()) => {
+									// Commit succeeded
+									let lqs: Vec<TrackedResult> =
+										txn.consume_pending_live_queries();
+									// Track the live queries in the data store
+									self.kvs.track_live_queries(&lqs).await?;
```

Comment:
Technically, anything after `commit()` may not run at all, when e.g. the node stopped immediately after returning from `commit()`.

Maybe having a code comment about why that is OK in this context would make it more easy to understand!

(I guess, the pending live queries are valid only within the node that the clients started the LQs? In other words, you don't need to deal with the case where another node to take over the LQ started on dead nodes. That might be the why?

Reply:
Ah, because this is always node-local. It does not relate to storage any more. Will add a comment 👍